framework/src/meshgenerators/ElementDeletionGeneratorBase.C

```diff
@@ -181,9 +195,10 @@ ElementDeletionGeneratorBase::generate()
       {
         const Elem * elem = mesh->elem_ptr(q.first);
         const unsigned int side = q.second;
-        const Elem * neighbor = elem->neighbor_ptr(side);
+        const Elem * target =
+            (side > elem->n_sides()) ? elem->interior_parent() : elem->neighbor_ptr(side);
```

Comment:
```suggestion
            (side >= elem->n_sides()) ? elem->interior_parent() : elem->neighbor_ptr(side);
```

Reply:
I think this change is correct?

Reply:
What the hell.  It's correct, and it's proof that my test case isn't testing as much as I'd hoped.  I guess I'd need to specifically cook up something where we're ghosting a boundary element but not ghosting its `interior_parent`?  Not sure how to force that; I'd been hoping that the tests I ran on my own (throwing that unit test at 64 processors to chop it up as finely as I could) would do it.

Reply:
You could write a custom `RelationshipManager` that you put in the `test` directory that does that